docs/snippets/number-of-products-per-row.md

```diff
@@ -0,0 +1,24 @@
+# Change number of related products output
+
+> This is a **Developer level** doc. If you are unfamiliar with code and resolving potential conflicts, select a [WooExpert or Developer](https://woocommerce.com/customizations/) for assistance. We are unable to provide support for customizations under our [Support Policy](http://www.woocommerce.com/support-policy/?).
+
+Add code to your child theme’s functions.php file or via a plugin that allows custom functions to be added, such as the [Code snippets](https://wordpress.org/plugins/code-snippets/) plugin. Avoid adding custom code directly to your parent theme’s `functions.php` file as this will be wiped entirely when you update the theme.
+
+Please note that it does not work for all themes because of the way they’re coded.
+
+```php
+if ( ! function_exists( 'YOUR_PREFIX_related_products_args' ) ) {
+	/**
+	 * Change number of related products output.
+	 *
+	 * @param  array $args The related products args.
+	 * @return array The modified related products args.
+	 */
+	function YOUR_PREFIX_related_products_args( $args ) {
+		$args['posts_per_page'] = 4; // 4 related products.
```

Comment:
Technically the return value of this filter can handle `array|string|object` because it is [passed through `wp_parse_args`](https://github.com/woocommerce/woocommerce/blob/69e9acaba98d2ab6a7eabe9d0e58d2df19c3b7ce/plugins/woocommerce/includes/wc-template-functions.php#L2068C1-L2068C1). So to be more resilient against other code hooking into this function that _might_ not be returning an array, client code should be testing for array first. Of course if some other code running in the site is _not_ returning an array, there would be larger problems, but at least this would prevent a fatal.
```suggestion
        If ( ! is_array( $args ) ) {
            $args = array();
        }
		$args['posts_per_page'] = 4; // 4 related products.
```